### DISCUSSION TITLE
Client side debug or use node tools

### DISCUSSION THREAD

**Discussion by mickeyvip** [livebud/bud#269] [Q&A]
Hello and thank you for this interesting project.

I am wondering 

- How we can debug the client-side code:
  - There seem to be no source maps
  - The "Svelte Dev Tools" does not recognize the client-side code as Svelte

- How we can use node tools such as:
  - postcss (for adding TailwindCSS, for example)

**Comment by matthewmueller** [comment]
Hey @mickeyvip thanks for your interest in Bud! It's still early days so a lot of obvious things still aren't available yet. 

> There seem to be no source maps

This is true, but should be pretty easy to add since ESBuild can easily generate them. We'd just need to alter the configuration in ESBuild, Svelte and expose them to the browser. PRs welcome!

> The "Svelte Dev Tools" does not recognize the client-side code as Svelte

Interesting. I haven't tried Svelte's devtools. Do you know why this would be?

> postcss (for adding TailwindCSS, for example)
 
I'm working on getting Tailwind working next. It'll be part of v0.3 or v0.4. Here's a discussion with a bit more context: https://github.com/livebud/bud/discussions/188.

---

If you're interested in contributing any of these features. Please let me know!

**Comment by mickeyvip** [comment]
Thank you for the quick response.

Looking at the Svelte Dev Tools, I see this:

> In order for svelte-devtools to comunicate with your application bundle the svelte compiler must have the dev option set to true.

Can that be that `bud` produces a **production** and not **dev** build of the Svelte app?

**Comment by mickeyvip** [comment]
I am not very proficient with Go, but looking at `bud/package/svelte/compiler.go` it seems like `Compiler.Dev` is hardcoded to be set to `true`.

So I am not sure what can be a problem.
